src/parsers/installrdf.js

```diff
@@ -1,66 +1,77 @@
-import { ADDON_TYPE_MAP, INSTALL_RDF, RDF_DEFAULT_NAMESPACE } from 'const';
-import { TYPE_INVALID, TYPE_MISSING } from 'messages';
+import { ADDON_TYPE_MAP, RDF_DEFAULT_NAMESPACE } from 'const';
+import { RDF_TYPE_INVALID, RDF_NAME_MISSING,
+         RDF_TYPE_MISSING } from 'messages';
 import log from 'logger';
-import RDFScanner from 'scanners/rdf';
 
 
 export default class InstallRdfParser {
 
-  constructor(rdfString, collector, {namespace=RDF_DEFAULT_NAMESPACE}={}) {
-    this.rdfString = rdfString;
+  constructor(xmlDoc, collector, {namespace=RDF_DEFAULT_NAMESPACE}={}) {
+    this.xmlDoc = xmlDoc;
     // Provides ability to directly add messages to
     // the collector.
     this.collector = collector;
     this.namespace = namespace;
   }
 
-  parseDoc() {
-    var rdfScanner = new RDFScanner(this.rdfString, INSTALL_RDF);
-    return rdfScanner.getContents();
-  }
-
   getMetaData() {
-    return this.parseDoc()
-      .then((xmlDoc) => {
```

Comment:
It felt wrong to me to need to keep passing the xmlDoc around. This meant this wasn't necessary. The `installrdf` metadata class' *only* job is to look things up from the XMLdoc. Returning a promise in a constructor is a no-no, otherwise that might have been a better solution. This meant that the doc is always available on the isntance and there's not a dependency on calling a specific method before calling another one as that happens externally.

Alot of our other code is doing explicit get something and pass it on down the chain. E.g. the files (metadata) from the xpi class.